examples/mnist-classification/src/mnist_classification/core.clj

```diff
@@ -0,0 +1,214 @@
+(ns mnist-classification.core
+  (:require [clojure.java.io :as io]
+            [cortex.datasets.mnist :as mnist]
+            [mikera.image.core :as i]
+            [think.image.image :as image]
+            [think.image.patch :as patch]
+            [think.image.data-augmentation :as image-aug]
+            [cortex.nn.layers :as layers]
+            [clojure.core.matrix.macros :refer [c-for]]
+            [clojure.core.matrix :as m]
+            [cortex.experiment.classification :as classification]
+            [cortex.experiment.train :as train]
+            [cortex.nn.network :as network]
+            [cortex.nn.execute :as execute]
+            [cortex.util :as util]
+            [cortex.experiment.util :as experiment-util])
+  (:import [java.io File]))
+
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+;; Convolutional neural net description
+(def image-size 28)
+(def num-classes 10)
+
+(def initial-description
+  [(layers/input 28 28 1 :id :data)
+   (layers/convolutional 5 0 1 20)
+   (layers/max-pooling 2 0 2)
+   (layers/dropout 0.9)
+   (layers/relu)
+   (layers/convolutional 5 0 1 50)
+   (layers/max-pooling 2 0 2)
+   (layers/batch-normalization)
+   (layers/linear 1000)
+   (layers/relu :center-loss {:label-indexes {:stream :labels}
+                              :label-inverse-counts {:stream :labels}
+                              :labels {:stream :labels}
+                              :alpha 0.9
+                              :lambda 1e-4})
+   (layers/dropout 0.5)
+   (layers/linear num-classes)
+   (layers/softmax :id :labels)])
+
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+;; Get Yann LeCun's mnist dataset and save it as folders of folders of png
+;; files. The top level folders are named 'training' and 'test'. The subfolders
+;; are named with class names, and those folders are filled with images of the
+;; appropriate class.
+(defn- ds-data->png
```

Comment:
Still feel that managing the dataset should be separate from training the network.